src/completions.ts

```diff
@@ -296,28 +296,17 @@ export abstract class CompletionSourceFuse extends CompletionSource {
     }
 
     /** Call to replace the current display */
-    // TODO: optionContainer.replaceWith and optionContainer.remove don't work.
-    // I don't know why, but it means we can't replace the div in one go. Maybe
-    // an iframe thing.
     updateDisplay() {
-        /* const newContainer = html`<div>` */
-
-        while (this.optionContainer.hasChildNodes()) {
-            this.optionContainer.removeChild(this.optionContainer.lastChild)
-        }
+        const newContainer = this.optionContainer.cloneNode(false)
 
         for (const option of this.options) {
-            /* newContainer.appendChild(option.html) */
             if (option.state !== "hidden")
-                this.optionContainer.appendChild(option.html)
+                // This is probably slow: `.html` means the HTML parser will be invoked
+                newContainer.appendChild(option.html)
```

Comment:
The slow part is the creation of the element in question, which happens in the constructor of `CompletionOptionHTML`'s descendants. This just accesses it.

Reply:
Yeah, you're right - https://github.com/tridactyl/tridactyl/blob/ced62d5fca6ea5aa986975cf5cf1765844498363/src/completions.ts#L102

I thought `.html` was a string that had to be parsed. Wasn't paying attention